vms/platformvm/reward_lock_tx.go

```diff
@@ -0,0 +1,191 @@
+	TxID ids.ID `serialize:"true" json:"txID"`
+}
+
+func (tx *UnsignedRewardLockTx) InitCtx(*snow.Context) {}
+
+func (tx *UnsignedRewardLockTx) InputIDs() ids.Set { return nil }
+
+func (tx *UnsignedRewardLockTx) SyntacticVerify(*snow.Context) error { return nil }
+
+// Attempts to verify this transaction with the provided state.
+func (tx *UnsignedRewardLockTx) SemanticVerify(vm *VM, parentState MutableState, stx *Tx) error {
+	_, _, err := tx.Execute(vm, parentState, stx)
+	return err
+}
+
+// Execute this transaction.
+//
+// The next lock to be removed must be have an end time equal to the current
+// chain timestamp.
+func (tx *UnsignedRewardLockTx) Execute(
+	vm *VM,
+	parentState MutableState,
+	stx *Tx,
+) (
+	VersionedState,
+	VersionedState,
+	error,
+) {
+	switch {
+	case tx == nil:
+		return nil, nil, errNilTx
+	case tx.TxID == ids.Empty:
+		return nil, nil, errInvalidID
+	case len(stx.Creds) != 0:
+		return nil, nil, errWrongNumberOfCredentials
+	}
+
+	lockState := parentState.LockChainState()
+	lockTx, lockReward, err := lockState.GetNextLock()
+	switch {
+	case err == database.ErrNotFound:
+		return nil, nil, fmt.Errorf("failed to get next lock stop time: %w", err)
+	case err != nil:
+		return nil, nil, err
+	}
+
+	lockTxID := lockTx.ID()
+	if lockTxID != tx.TxID {
+		return nil, nil, fmt.Errorf(
+			"attempting to unlock TxID: %s. Should be unlocking %s",
+			tx.TxID,
+			lockTxID,
+		)
+	}
+
+	// Verify that the chain's timestamp is the validator's end time
+	currentTime := parentState.GetTimestamp()
+	lockTimedTx, ok := lockTx.UnsignedTx.(TimedTx)
+	if !ok {
+		return nil, nil, errWrongTxType
+	}
+	if endTime := lockTimedTx.EndTime(); !endTime.Equal(currentTime) {
+		return nil, nil, fmt.Errorf(
+			"attempting to unlock TxID: %s before their end time %s",
+			tx.TxID,
+			endTime,
+		)
+	}
+
+	newlyLockState, err := lockState.DeleteNextLock()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentStakerState := parentState.CurrentStakerChainState()
+	pendingStakerState := parentState.PendingStakerChainState()
+
+	onCommitState := newVersionedState(parentState, currentStakerState, pendingStakerState, newlyLockState)
+	onAbortState := newVersionedState(parentState, currentStakerState, pendingStakerState, newlyLockState)
+
+	// If the reward is aborted, then the current supply should be decreased.
+	currentSupply := onAbortState.GetCurrentSupply()
+	newSupply, err := math.Sub64(currentSupply, lockReward)
+	if err != nil {
+		return nil, nil, err
+	}
+	onAbortState.SetCurrentSupply(newSupply)
+
+	switch uLockTx := lockTx.UnsignedTx.(type) {
+	case *UnsignedAddLockTx:
+		// Refund the stake here
+		for i, out := range uLockTx.LockedOuts {
+			utxo := &avax.UTXO{
+				UTXOID: avax.UTXOID{
+					TxID:        tx.TxID,
+					OutputIndex: uint32(len(uLockTx.Outs) + i),
+				},
+				Asset: avax.Asset{ID: vm.ctx.AVAXAssetID},
+				Out:   out.Output(),
+			}
+			onCommitState.AddUTXO(utxo)
+			onAbortState.AddUTXO(utxo)
```

Comment:
Yep, just no reward in case of abort
Had the question about what should happen in case of abort, but its lost somewhere and never got to the team, I think
You can see in stakers code, that they also do the same in case of abort - funds unlocked, but no reward